### ISSUE TITLE
New interface for heterogeneity and demographic analysis

### ISSUE THREAD

**Issue by pierucci** [pierucci/heemod#110]
Following suggestions from @MattWiener, I propose a new interface for demographic and heterogeneity analysis. The model would be re-fit using the generic `update()` function (whose title is *Update and Re-fit a Model Call*).

```r
res <- run_models(...)

new <- update(res, newdata = tab_new)
```

And the `print` method on the computed object would display both the results of demographic and heterogeneity analysis (non-combined and combined). The `plot`  method would depend on the `type` argument, e.g. where `counts` would display the classical count plot of a `run_models()` object.

In conclusion:

  1. The calculations would only run once.
  2. Heterogeneity and demographic analysis would end up mixed together.
